Library/Homebrew/utils.sh

```diff
@@ -0,0 +1,47 @@
+puts_with_style() {
+  local message output_fd="$1" plain="$2" styled="$3"
+  shift 3
+
+  if [[ $# -eq 0 ]]
+  then
+    IFS= read -d '' -r message # Read message from `stdin`.
+    message="${message%$'\n'}" # Trim trailing newline.
+  else
+    message="$*"
+  fi
+
+  if [[ -t "$output_fd" ]] # If FD is a TTY, style message with color/bold/...
+  then
+    # shellcheck disable=SC2059
+    printf "$styled\n" "$message"
+  else
+    # shellcheck disable=SC2059
+    printf "$plain\n" "$message"
+  fi
+}
+
+ohai() {
+  # Use bold and blue color if `stdout` is a TTY.
+  puts_with_style 1 "==> %s" "\033[1;34m==>\033[1;39m %s\033[0m" "$@"
```

Comment:
It should become obvious when looking at the first line of `puts_with_style` where the first argument gets assigned to `output_fd`. Would it be sufficiently obvious if I modified the comment above this line to `Use bold and blue color if FD 1 (stdout) is a TTY.`?